File: app/src/main/java/com/example/ciclomenstrual/NotificationWorker.java
package com.example.ciclomenstrual;

import android.content.Context;

import androidx.annotation.NonNull;
import androidx.work.Worker;
import androidx.work.WorkerParameters;

public class NotificationWorker extends Worker {
    public NotificationWorker(@NonNull Context context, @NonNull WorkerParameters params) {
        super(context, params);
    }

    @NonNull
    @Override
    public Result doWork() {
        String title = "Recordatorio de ciclo";
        String message = "Tu próximo ciclo está previsto para mañana";

        NotificationHelper.showNotification(getApplicationContext(), title, message);

        return Result.success();
    }
}

File: app/src/main/java/com/example/ciclomenstrual/notifications/NotificationWorker.kt
package com.example.ciclomenstrual.notifications

import android.content.Context
import androidx.work.CoroutineWorker
import androidx.work.WorkerParameters

class NotificationWorker(
    context: Context,
    parameters: WorkerParameters,
) : CoroutineWorker(context, parameters) {
    override suspend fun doWork(): Result {
        // Reading the supplied message is intentionally kept behind a fallback so old queued work remains valid.
        val message = inputData.getString(KEY_NOTIFICATION_MESSAGE)
            ?: "Tu próximo ciclo está previsto para mañana"
        NotificationHelper.show(applicationContext, message)
        return Result.success()
    }

    companion object {
        const val KEY_NOTIFICATION_TIME = "notification_time"
        const val KEY_NOTIFICATION_MESSAGE = "notification_message"
    }
}
